packages/client/src/comp/Address/Fields.tsx

```diff
@@ -0,0 +1,157 @@
+  .mui-select label { overflow: hidden; }
+`
+
+// Since react-hooks/exhaustive-deps will trigger re-renders due to its
+// depedencies, we declare this function outside of our component.
+const useDidMount = (fun: () => void) => React.useEffect(fun, [])
+
+export const AddressFields: React.FC<Props> = ({ type }) => {
+  const {
+    fields: baseAddressFields,
+    mailingFields,
+    setField: setBaseAddressField,
+    setMailingField,
+  } = AddressInputPartContainer.useContainer()
+  const { path } = useAppHistory()
+  const addrRef = useControlRef<Input>()
+  const fields = type === 'initialAddress' ? baseAddressFields : mailingFields
+  const setField = type === 'initialAddress' ? setBaseAddressField : setMailingField
+  const { fetchingData } = FetchingDataContainer.useContainer()
+  // When we first arrive at page, set focus and move cursor to beginning
+  useDidMount(() => {
+      if (type === 'initialAddress') {
+        if (path?.type === 'address' && addrRef?.current) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const controlEl = (addrRef.current as any).controlEl as HTMLInputElement
+          controlEl.focus({preventScroll: true})
+          controlEl.setSelectionRange(0, 0)
+        }
+      }
+  })
+
+  return <FlexBox>
+    {/* We use a 7:3 basis proportion between Flex Components */}
+    {/* The spacing between them 2% of the basis */}
+
+    <Flex basis='70%'>
+      <BaseInput
+        id='addr-street-input'  // This id is used for Warning Box to fill form quickly
+        label='Street Address'
+        ref={addrRef}
+        required
+        translate='no'
+        lang='en'
+        value={fields?.street}
+        onChange={e => setField('street', e.currentTarget.value)}
+      />
+    </Flex>
+
+    <Flex basis='28%'>
+      <BaseInput
+        label='Apartment'
+        id='addr-apt-input'
+        translate='no'
+        lang='en'
+        value={fields?.unit}
+        onChange={e => setField('unit', e.currentTarget.value)}
+      />
+    </Flex>
+
+    <Flex basis='40%'>
+      <BaseInput
+        id='addr-city-input'  // This id is used for Warning Box to fill form quickly
+        label='City'
+        required
+        translate='no'
+        lang='en'
+        value={fields?.city}
+        onChange={e => setField('city', e.currentTarget.value)}
+      />
+    </Flex>
+
+    <Flex basis='28%' className='mui-select'>
+      <Select
+        id='addr-state-input'  // This id is used for Warning Box to fill form quickly
+        label='State'
+        translate='no'
+        lang='en'
+        onChange={e => {
+          // MuiCSS has a buggy support for <Select/> when using TypeScript,
+          // to really access HTMLSelect and its value we to do this hack
+          const trueSelect = e.currentTarget.firstChild as HTMLSelectElement
+          setField('state', trueSelect.value)
+        }}
+        defaultValue={fields?.state}
```

Comment:
Do you use defaultValue or `<Option selected/>`?

Reply:
I'd like to use only value at `<Select/>`, unfortunately MuiCSS with TypeScript does not support it, so I'm using `defaultValue` since when the state changes a re-render is going to happen and defaultValue is equal to the state again.